### DISCUSSION TITLE
Managing cable changes

### DISCUSSION THREAD

**Discussion by tfboy** [netbox-community/netbox#10616] [Q&A]
Quick question: how do people go about managing cable changes?

In my mind, allowing a secondary connection with a status of "Planned" (or "Decommissioning") in addition to "Connected" would be great. That way, I could have the normal currently-connected cable going from equipment A to equipment B, but also document the upcoming cable going from equipment A to equipment C.
Currently, the model does not allow this as you can only have one connection per port / interface.
I'm not suggesting this changes as I guess that would break so many things.

In a typical scenario, we would like to plan a dozen or so cable changes, have them highlighted or searchable somehow within Netbox so that during a maintenance period, a shortlist could be easily generated and processed. The old "Connected" cables would be updated / deleted to match the new connection points.

So how does one manage the change?

P.S. Thank you for adding the ability to update an end point without having to delete the cable and recreate a new one. That's already a big improvement in my eyes :)

**Comment by candlerb** [comment]
The same applies to moving devices around in racks: you can't put a "planned" device in a rack location currently occupied by another device, but which you know is going to be removed.  Conversely, a single device can't simultaneously have a "current" location and a new "planned" location - which is something I would have found extremely helpful when managing a large data centre move. I ended up planning all the new locations in spreadsheets, and then only updating Netbox after each device was moved to its new home.

I think the same applies with cables, on a smaller scale.  If it's a replacement or move of one end, rather than installation of a completely new cable, then you'd have to plan it outside of Netbox, and update Netbox after the fact.

I haven't played with the multi-connected cable endpoints in Netbox 3.3 yet. Perhaps it could be abused to show one leg currently connected and one leg planned.  Even then, the "planned" leg couldn't connect to a port which is currently occupied by a cable termination.

Otherwise, I think you might be have to use comments or custom fields on cables to implement your desired workflow.

**Reply by tfboy** [reply]
Thanks Brian,
What I've done for now is create a second set of network interfaces and assigned the new cable to it, marked it as planned.
I'll then delete the cables of the existing NICs post move and reassign the cable end point on the kit to the main NICs and then delete the temp NICs.
That way I can search for planned cables and have a shortlist I can send to the wiring team.

Cheers!

**Comment by PieterL75** [comment]
Looks like a nice milestone feature.
Both the dual cable option and the reservation thingy...
I'm fond of the idea

**Comment by bluikko** [comment]
I have sometimes thought that it would be nice if it was possible to generate "change requests" from NetBox. It could take care of the OP as well. For example it could work something like:
- Click "Start change" button.
- Do the changes that need to be done, for example delete cableA from devA/p1 to devB/p1 and create cableB from devA/p2 to devC/p1.
- Click "End change" button.
- Somehow (figuring this out is the big question I guess...) record those changes and keep them in a separate "planned" section as changeset1 for example and flagged as so.
- Going to display the changeset1 could be printed/copy-pasted for doing the change step by step.
- While displaying the changeset click button "Commit change" to mark it as done and commit the changes to NetBox.

Obviously this would need colossal amount of more planning and so on but it would be interesting to hear if other people would be interested in such a functionality.  
And above all, how possible would it even be to make something like this to happen?

**Reply by xkilian** [reply]
I think these are two different things. You should manage your change workflow seperately from the creation of the planned cables, as they may not be put in service at the same time.

1. tfboy proposal permits a way to independantely add planned cables and the meta data provides links to what project/billing/whatever custom fields/tags to sort filter group these planned cables.
2. The switch over would be simple with what is proposed. And cleanup happens after implementation (which is sub-optimal, but makes more sense from a planning perspective)
3. #10851 Should handle mass changes using automation that are applied in a general perspective. Could be used to create the cables or even switch from planned to connected.
4.  External change management system should be used to track the change process. That could reference what is being done for #10851 or a general description. Not perfect. Changes can be viewed in the change logs for forensics if required.

**Comment by mtinberg** [comment]
That kind of change management is something that Nautobot was working on, converting the DB to MySQL and using a MySQL-specific database-versioning tool to provide the workflow you describe.  I don't know how well this works in practice, it's a neat trick if it does.

For proscribed changes that take multiple steps you could use a Script to make the changes happen altogether, for other changes which require engineering judgement sometimes you can use status fields to make your automation tools ignore changes which are in-progress, using process/procedure to make the intended state clear, when it's not something you are immediately deploying to equipment.

I think Netbox is built around the idea that changes are automated, so it's not trying to support the use of step-by-step instructions, however some changes are to physical equipment (cables, racks), maybe you could make a workflow around the ChangeLog to show adds/removes over a certain time frame to make a printout for field staff with the adds/changes/removes.

—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by PieterL75** [comment]
@tfboy I'd suggest to submit a feature request for this, you have my vote !
We recently moved all our cabling from another system to netbox, and I have my first cabling project that involves moving cables around, and not only new cables.

This idea of having multiple cables on one port, where only one can has the 'Active' status and is used for traces... I love it

**Comment by tfboy** [comment]
Well, I've tried to put in a feature request for this: https://github.com/netbox-community/netbox/issues/10882
Please go easy on me, first time I'm attempting this :)

**Reply by fatman00** [reply]
Has the FR gotten any response? This is a feature I am missing as well.

**Reply by TorstenDrees** [reply]
@tfboy: thank you for creating the FR. Sadly the FR was closed due to inactivity in Feb 2023. I am missing this kind of feature too. Is it possible for you to re-enable that FR (if you are still interested in it?)

**Reply by xkilian** [reply]
I still think this is a great FR, but the main issue is getting a developper interested in supporting this. Netbox has or other dev.
If this is interesting to you you should thumbs up the fr or the new fr.
